flink-table/flink-table-planner-blink/src/main/java/org/apache/flink/table/api/ConfigurationUtils.java

```diff
@@ -0,0 +1,123 @@
+
+	/**
+	 * Updates configuration with a specified minimum and a maximum time interval for how long idle state, i.e., state
+	 * which was not updated, will be retained.
+	 * State will never be cleared until it was idle for less than the minimum time and will never
+	 * be kept if it was idle for more than the maximum time.
+	 * When new data arrives for previously cleaned-up state, the new data will be handled as if it
+	 * was the first data. This can result in previous results being overwritten.
+	 * Set to 0 (zero) to never clean-up the state.
+	 *
+	 * @param minTime The minimum time interval for which idle state is retained. Set to 0 (zero) to
+	 * never clean-up the state.
+	 * @param maxTime The maximum time interval for which idle state is retained. May not be smaller
+	 * than than minTime. Set to 0 (zero) to never clean-up the state.
+	 * @param configuration configuration to update.
+	 */
+	public static Configuration withIdleStateRetentionTime(Configuration configuration, Time minTime, Time maxTime) {
+		configuration.setLong(SQL_EXEC_STATE_TTL_MS, minTime.toMilliseconds());
+		configuration.setLong(SQL_EXEC_STATE_TTL_MAX_MS, maxTime.toMilliseconds());
+		return configuration;
+	}
+
+	/**
+	 * Gets the maximum time until state which was not updated will be retained.
+	 *
+	 * @param configuration the configuration object
+	 * @return the maximum time until state which was not updated will be retained.
+	 */
+	public static Long getMinIdleStateRetentionTime(Configuration configuration) {
+		return configuration.getLong(SQL_EXEC_STATE_TTL_MS);
+	}
+
+	/**
+	 * Returns the maximum time until state which was not updated will be retained.
+	 *
+	 * @param configuration the configuration object
+	 * @return the maximum time until state which was not updated will be retained.
+	 */
+	public static Long getMaxIdleStateRetentionTime(Configuration configuration) {
+		if (configuration.contains(SQL_EXEC_STATE_TTL_MS) && !configuration.contains(SQL_EXEC_STATE_TTL_MAX_MS)) {
+			configuration.setLong(SQL_EXEC_STATE_TTL_MAX_MS, getMinIdleStateRetentionTime(configuration) * 2);
+		}
+		return configuration.getLong(SQL_EXEC_STATE_TTL_MAX_MS);
+	}
+
+	/**
+	 * Returns whether a operator is disabled.
+	 *
+	 * @param configuration the configuration object
+	 * @return return true if the operator is disabled, else false.
+	 */
+	public static boolean isOperatorDisabled(Configuration configuration, DisabledOperatorType operatorType) {
+		String disabledOperators = configuration.getString(TableConfigOptions.SQL_EXEC_DISABLED_OPERATORS);
+		String[] splittedDisabledOperators = disabledOperators.split(",");
+		Set<DisabledOperatorType> disabledOperatorSet = new HashSet<>();
+		for (String splittedDisabledOperator : splittedDisabledOperators) {
+			splittedDisabledOperator = splittedDisabledOperator.trim();
+			if (splittedDisabledOperator.isEmpty()) {
+				continue;
+			}
+			if (splittedDisabledOperator.equals("HashJoin")) {
```

Comment:
use `DisabledOperatorType` ?

Reply:
If config 'HashJoin', means to disable 'BroadcastHashJoin'  and 'ShuffleHashJoin'.